ui/src/components/query/viewer/RecursiveMatchDisplay.vue

```diff
@@ -0,0 +1,169 @@
+        v-if="isArrayHasLength(match.where)"
+        v-for="(nestedWhere,index) in match.where"
+        :property="nestedWhere"
+        :depth = "depth"
+        :property-index="index"
+        :key="index"
+        :second-property="index === 1"
+    />
+    <RecursiveMatchDisplay
+        v-if="match.then"
+        :match="match.then"
+        :inline="false"
+        :depth="1"
+    />
+
+
+
+    <div v-if="match.variable" :class="'variable-field'" :style="{ paddingLeft: depth * 3+2 + 'rem' }">
+      <span>named {{match.variable}}</span>
+    </div>
+    <span v-if="match.match" :style="{ paddingLeft: (depth * 3) + 'rem' }">)</span>
+   </component>
+</template>
+
+
+<script setup lang="ts">
+import { isArrayHasLength } from "@im-library/helpers/DataTypeCheckers";
+import { Match,Assignable,IriLD,Node} from "@im-library/interfaces/AutoGen";
+import { onMounted, Ref, ref,watch,defineProps } from "vue";
+import RecursiveWhereDisplay from "./RecursiveWhereDisplay.vue";
+import QueryOverlay from "./QueryOverlay.vue";
+
+
+
+interface Props {
+  match: Match;
+  isVariable? : boolean;
+  depth: number;
+  inline : boolean;
+
+}
+
+const props = defineProps<Props>();
+const op: Ref<any> = ref();
+const clickedNodeRef: Ref<Match> = ref({} as Match);
+const list: Ref<Match[]> = ref([]);
+const op1: Ref<any> = ref();
+
+
+function indentationStyle(inline:boolean, depth:number){
+  return {
+    paddingLeft: !inline ? depth * 2 + 'rem' : '0rem'
+  }
+}
+function getFormattedNodes(values: Node[]) {
+  const value= values
+      .map(item => {
+        const parts = [];
+        if (item.qualifier)
+          parts.push("<span>"+item.qualifier+"</span>");
+        if (item.name)
+          parts.push("<span style=\"color : rgb(0,102,102);\">"+item.name+"</span>");
+        return parts.join(' ');
+      })
+      .join(', ');
+    return value;
+}
+
+function getFormattedPath(paths: IriLD[]) {
+  const path= paths
+      .map(item => {
+        const parts = [];
+        if (item.qualifier)
+          parts.push("<span>"+item.qualifier+"</span>");
+        if (item.name)
+          parts.push("<span style=\"color : rgb(0,102,102);\">"+item.description+"</span>");
+        return parts.join(' ');
+      })
+      .join('-> ');
+  return path;
+}
+// Watch for changes in the prop and update the local copy accordingly
+
+
+</script>
+
+
+<style scoped>
+.feature {
+  display: flex;
+  flex-flow: column;
+  margin-left: 1rem;
+  margin-top: 0.1rem;
+  margin-bottom: 0.1rem;
+}
+
+.feature-indent {
+  display: flex;
+  flex-flow: column;
+  margin-left: 1rem;
+  margin-top: 0.1rem;
+  margin-bottom: 0.1rem;
+}
+.then {
+  padding-right: 0.2rem;
+  color : rgb(72,32,92)
+}
+.field {
+  padding-right : 0.2rem;
+}
+
+.variable-field {
+  padding-left : 2rem;
+  color : rgb(180, 48, 190);
+  padding-right : 0.3rem;
+}
+
+
+.return {
+  color: lightseagreen;
+  padding-left: 0.5rem;
+}
+
+.output {
+  color: mediumslateblue;
+}
+</style>
+
+<style>
```

Comment:
replace global style sheet with :deep styles in scoped section and use theme colors
https://vuejs.org/api/sfc-css-features.html